assets/dir-index-html/src/script.js

```diff
@@ -0,0 +1,22 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const listingsPath = [
+        {{ range .Listing }}
+            '{{.Path}}',
+        {{ end }}
+    ];
+
+    const init = {
+        method: 'GET',
+        headers: {'Accept': 'application/vnd.ipfs.cache.local'}
+    };
+
+    const notCachedHTML = '<div title="File not cached locally" class="ipfs-_attention">&nbsp;</div>';
+
+    listingsPath.forEach((element, index) => {
+        fetch(element, init)
```

Comment:
AFAIK this request the full file, we dont want to download GiBs just because you openned the dir listing.

Reply:
- Even with inexpensive `HEAD`  `application/vnd.ipfs.cache*` check we don't want to send 10k requests when opening big directory like bafybeiggvykl7skb2ndlmacg2k5modvudocffxjesexlod2pfvg5yhwrqm 
   - Even 1k `QmdmQXB2mzChmMeKY47C43LxUdg1NDJ5MWcKMKxDu7RgQm` will be wasteful, if only a small subset is visible on the screen.

This should be refactored to fetch status only for visible items – see https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API